ci: minor improvements

blueyed opened the pull request.

blueyed committed:
ci/common/test.sh: fix some issues reported by shellcheck

justinmk commented on ci/common/test.sh:
```diff
@@ -131,8 +132,7 @@ check_runtime_files() {(
+  for file in $(git -C runtime ls-files "$@") ; do
     # Check that test is not trying to work with files with spaces/etc
     # Prefer failing the build over using more robust construct because files
     # with IFS are not welcome.
```
I don't think `file` has the full/absolute path, so the `test -e ...` fails.

```
E /tests/install_nvim|runtime-install :: It appears that autoload/RstFold.vim is only a part of the file name
E /tests/install_nvim|runtime-install :: It appears that autoload/ada.vim is only a part of the file name
E /tests/install_nvim|runtime-install :: It appears that autoload/adacomplete.vim is only a part of the file name
E /tests/install_nvim|runtime-install :: It appears that autoload/ccomplete.vim is only a part of the file name
E /tests/install_nvim|runtime-install :: It appears that autoload/clojurecomplete.vim is only a part of the file name
```

Rather than work around this, it's better just to keep the `cd runtime`. Using subshells is very common in various shell scripts throughout the project.

blueyed replied:
Oh, I've missed that it was using a subshell in the function now..

Will remove this commit here then.  Changing to a WIP for minor things.

blueyed committed:
ci: AppVeyor: use fast_finish=true

blueyed committed:
ci: Travis: remove obsolete cmake file

Added in f65614fb1, forgot to be removed in 55682710a.

blueyed committed:
ci/build.ps1: add comment for PATH mangling with old tests

blueyed commented:
I could take https://github.com/neovim/neovim/pull/10226 into here also (or push it together), for one less build when merging.

blueyed committed:
ci: Travis: ccache: use --zero-stats

This makes it display actual results in before_cache then, and it seems
like it is not necessary to keep stats across builds.

blueyed committed:
ci: Travis: skip lint job with "\[skip.lint\]"

blueyed commented:
Added handling of "[skip-lint]" (worked: https://travis-ci.org/neovim/neovim/builds/546007291).
